src/api/useViewport.js

```diff
@@ -0,0 +1,85 @@
+import { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { setViewportFeatures } from '../redux/cartoSlice';
+import bboxPolygon from '@turf/bbox-polygon';
+import intersects from '@turf/boolean-intersects';
+
+const GEOMETRY_TYPES = Object.freeze([
+  'Point',
+  'MultiPoint',
+  'LineString',
+  'MultiLineString',
+  'Polygon',
+  'MultiPolygon'
+]);
+
+// TODO: need to find another way to get the unique feature id
+// User will need to specify the unique id as a widget prop ?¿
+// useViewport receives another argument with unique id ?¿
+function uniqueId(feature) {
```

Comment:
we should be able to get this somehow from the layer, as it's a param the user could set. See `uniqueIdProperty` at https://deck.gl/docs/api-reference/carto/carto-sql-layer

Reply:
Indeed, but we should make this deck.gl prop required in our layers right? User will need to specify this always

Reply:
@AdriSolid This should be kept optional and have a good default value per layer type. Right now, CartoSQLLayer already has one, see: https://github.com/visgl/deck.gl/blob/42c2270289188c0a04f4cea1e7d69ba7a5d484be/modules/carto/src/layers/carto-sql-layer.js#L8 and that is what needs to be used, from the layer. That param has to be honoured.

For the BQTilerLayer, I'm not so sure about current defaults on deck.gl side. That should be checked and honoured the same way.

Reply:
As far as I know, BQTilerLayer does not have a default (only DO layers..). My proposal is to add a second required argument to `useCartoLayerFilterProps` with the unique id field.

Template
```javascript
function StoresLayer() {
  const cartoFilterProps = useCartoLayerFilterProps(source, 'myUniqueId');
...
```

Lib
```javascript
export default function useCartoLayerFilterProps(source, uniqueId = 'cartodb_id') {
    const [onViewportLoad] = useViewportFeatures(source, uniqueId);
    ...
};
```

What do you think? @VictorVelarde